misc/python/materialize/cloudtest/exists.py

```diff
@@ -0,0 +1,39 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+
+import subprocess
+
+from materialize import ui
+from materialize.ui import UIError
+
+
+def exists(resource: str, context: str = "kind-kind") -> None:
+    __exists(resource, True, context)
+
+
+def not_exists(resource: str, context: str = "kind-kind") -> None:
+    __exists(resource, False, context)
+
+
+def __exists(resource: str, should_exist: bool, context: str = "kind-kind") -> None:
+    cmd = ["kubectl", "get", "--output", "name", resource, "--context", context]
+    ui.progress(f'running {" ".join(cmd)} ... ')
+
+    try:
+        result = subprocess.run(cmd)
+        result.check_returncode()
+        if should_exist:
+            ui.progress("success!", finish=True)
+        else:
+            raise UIError(f"{resource} exists, but expected it not to")
+    except subprocess.CalledProcessError:
```

Comment:
This is a little scary because this will return success if the resource is expected to not exist but `kubectl` returns _any_ error. Might just want to check whether "(NotFound)" is in the output of `kubectl`? A bit gross, but it should be safe enough in practice.